src/physics/reynolds_averaged_navier_stokes.cpp

```diff
@@ -0,0 +1,823 @@
+{   
+    std::array<real2,dim+2> conservative_soln_rans;
+    for(int i=0; i<dim+2; i++){
+        conservative_soln_rans[i] = conservative_soln[i];
+    }
+ 
+    return conservative_soln_rans;
+}
+//To do later
+//Introducing new template to build general extract function for RANS and turbulence model
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+template <typename real2>
+std::array<dealii::Tensor<1,dim,real2>,dim+2> ReynoldsAveragedNavierStokesBase<dim,nstate,real>
+::extract_rans_solution_gradient (
+    const std::array<dealii::Tensor<1,dim,real2>,nstate> &solution_gradient) const
+{   
+    std::array<dealii::Tensor<1,dim,real2>,dim+2> solution_gradient_rans;
+    for(int i=0; i<dim+2; i++){
+        solution_gradient_rans[i] = solution_gradient[i];
+    }
+ 
+    return solution_gradient_rans;
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+template <typename real2>
+std::array<dealii::Tensor<1,dim,real2>,nstate-(dim+2)> ReynoldsAveragedNavierStokesBase<dim,nstate,real>
+::dissipative_flux_turbulence_model (
+    const std::array<real2,dim+2> &primitive_soln_rans,
+    const std::array<real2,nstate-(dim+2)> &primitive_soln_turbulence_model,
+    const std::array<dealii::Tensor<1,dim,real2>,nstate-(dim+2)> &primitive_solution_gradient_turbulence_model) const
+{   
+    std::array<real2,nstate-(dim+2)> effective_viscosity_turbulence_model; 
+
+    if constexpr(std::is_same<real2,real>::value){ 
+        effective_viscosity_turbulence_model = compute_effective_viscosity_turbulence_model(primitive_soln_rans, primitive_soln_turbulence_model);
+    }
+    else if constexpr(std::is_same<real2,FadType>::value){ 
+        effective_viscosity_turbulence_model = compute_effective_viscosity_turbulence_model_fad(primitive_soln_rans, primitive_soln_turbulence_model);
+    }
+    else{
+        std::cout << "ERROR in physics/reynolds_averaged_navier_stokes.cpp --> dissipative_flux_turbulence_model(): real2!=real || real2!=FadType)" << std::endl;
+        std::abort();
+    }
+
+    std::array<dealii::Tensor<1,dim,real2>,nstate-(dim+2)> viscous_flux_turbulence_model;
+
+    for(int i=0; i<nstate-(dim+2); i++){
+        for(int j=0; j<dim; j++){
+            viscous_flux_turbulence_model[i][j] = -effective_viscosity_turbulence_model[i]*primitive_solution_gradient_turbulence_model[i][j];
+        }
+    }
+    
+    return viscous_flux_turbulence_model;
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+template <typename real2>
+std::array<real2,nstate-(dim+2)> ReynoldsAveragedNavierStokesBase<dim,nstate,real>
+::convert_conservative_to_primitive_turbulence_model (
+    const std::array<real2,nstate> &conservative_soln) const
+{   
+    std::array<real2,nstate-(dim+2)> primitive_soln_turbulence_model;
+    for(int i=0; i<nstate-(dim+2); i++){
```

Comment:
`(dim+2)` is being used often throughout this file, consider declaring a `static const double dim2 = dim+2;` in the header file; see `src/flow_solver/flow_solver_cases/periodic_turbulence.h`

Reply:
in addition, create another static const variable for `nstate-nstate_navier_stokes`